### ISSUE TITLE
Module Input File Syntax

### ISSUE THREAD

**Issue by friedmud** [idaholab/moose#5585]
It's time to make a plan for Module syntax.  So far it's just been a free-for-all when it comes to deciding where to put custom syntax for Modules.  We have some that are main blocks ( `SolidMechanics`).  Some that are permanent sub-blocks (`Kernels/AxisymmetricRZ`) and other random stuff all over the place.

This presents a confusing and cluttered interface to users.  Peacock just looks like a damn mess.

Here's what I propose:

```
[Modules]
  [./TensorMechanics]
    [./AxisymmetricRZ]
    [../]
  [../]
  [./SolidMechanics]
    stuff = junk
  [../]
  [./PhaseField]
    otherstuff = more_junk
  [../]
[]
```

Basically: each Module gets a sand-box underneath the main `[Modules]` block where you can put whatever custom syntax you want.

Questions?  Comments?

**Comment by permcody** [comment]
Tagging @jasondhales, @tonkmr, @dschwen, @idaholab/moose-developers

**Comment by smharper** [comment]
I don't like how that would split up the `Kernels` block
```
[Kernels]
  [./PP]
    type = MyOwnSpecialKernel
    variable = temperature
  [../]
[]
...
[Modules]
  [./HeatConduction]
    [Kernels]
      [./conduction]
        type = HeatConduction
        variable = temperature
      [../]
    []
  [../]
[]
```

**Comment by YaqiWang** [comment]
I do not know why, modules are based on MOOSE, they will create new syntax without any conflict with the existing MOOSE syntax. I am sure module developers will be fine if MOOSE decided to take over some syntax used by them.

**Comment by friedmud** [comment]
@smharper That's not the idea.  Normal Kernels, etc. would still be put where they normally are.  This is for the Custom Actions that are in Modules.

For instance... look at https://github.com/idaholab/moose/blob/devel/modules/tensor_mechanics/tests/2D_geometries/2D-RZ_finiteStrain_test.i#L59

It has:

```
[Kernels]
  [./AxisymmetricRZ]
    use_displaced_mesh = true
  [../]
[]
```

What is up with that?  It is just completely wrong.  There is no `type`?  I actually thought for a moment that the `type` might just be inferred from the name of the block (which is the name of a class... which doesn't match anything else in MOOSE)... but NO... this is just a custom Action... it's just very out of place!

If you open up a Peacock window with a new Stork based application and expand "Kernels" you see a big mess of permanent custom syntax sub-blocks that belong to different Modules.

**Comment by dschwen** [comment]
We could also "namespace" action underneath MOOSE top-level blocks. So we'd
have Kernels/TensorMechanics/* and Variables/PhaseField/* etc.
That would keep objects of the same kind closer together.

**Comment by aeslaughter** [comment]
I think I am with @friedmud, all Phase-field related stuff should be under a top-level modules

Here is what I am planning for the level-set module:

```
[Modules]
  [./Levelset] # the block existence enables the complete, base levelset equation
    variable = phi
    supg = true # turns on SUPG kernels
    gls = true # enables the Galerkin Least Square stabilization
    volume = true # creates AuxVariable, AuxKernel, and Postprocessor for computing volume of region
    cfl_condition # creates Postprocessors and setups Executioner to use CFLcondition PP for timesteps
    ...
  [../]
[]
```

**Comment by dschwen** [comment]
Yeah, I have chatted with @friedmud in the mean time and he did convince me. This will however be a painful change. But I guess as our input files are easily parseable we (you) could provide a converter to move the old actions to the new hierarchy.

**Comment by friedmud** [comment]
It's easy to leave the old syntax for a while too.  We can deprecate the old syntax for as long as it makes sense.

What is the current status on the switch from solid_mechanics to tensor_mechanics?  There may not be a need to screw with the `[SolidMechanics]` block... only to have it go away soon.  So we could just leave that alone for now.

**Comment by permcody** [comment]
While this is a nice to have, I don't think it is a high priority for us even now.

**Comment by GiudGiud** [comment]
I think this can be closed?
It's like this now
```
[NavierStokes]

[]
```
